src/Core/Foundation/Filesystem/FileSystem.php

```diff
@@ -30,6 +30,11 @@
 
 class FileSystem
 {
+    /**
+     * Default mode for directories and files
+     */
+    const DEFAULT_MODE = 0755;
```

Comment:
The default mode for files and folder should not be the same:
- folder: 755
- files: 644

So you should have two different const `FOLDER_DEFAULT_MODE` and `FILE_DEFAULT_MODE` (even though you used this const only once for a file It's safer to differentiate both modes).

Besides, depending on your server config sometimes you need to have more open permissions, I'm thinking 775 and 664 permissions (for cases where your server user doesn't match your cli user for example). In which case it is convenient to have a common group and allow writing for this group.

So maybe this modes should be editable in the back office and we could use static getters in this class:

```
public static function getFolderDefaultMode() {
    return Configuration::get('PS_FOLDER_DEFAULT_MODE', self::FOLDER_DEFAULT_MODE);
}

public static function getFileDefaultMode() {
    return Configuration::get('PS_FILE_DEFAULT_MODE', self::FILE_DEFAULT_MODE);
}
```

Maybe it's overkill for the scope of this PR but I think it would be really nice to have and it would prevent us to break some customers environment by allowing them to override this modification.

Reply:
You're right, I add the change for 0644 ;) But it's a little overkill to add this in configuration, more because many developers don't know how it works, so it'll add a real complexity for merchants.

Reply:
Well if they don't know how it works they won't look for the setting anyway :P
But if they are stuck because of our modif we would have at least a way to help them fix the issue through the admin
I thought adding these two fields in `Advanced Parameters -> Administration` wouldn't be that long to implement, but it could really save us time for support later

Reply:
Everything else seems fine, this is the last feedback to deal

As I read the previous conversation I saw @colinegin also proposed to have a way to bypass this restriction. With the current modification the only possibility is to  modify the const in `Filesystem` (which is not overridable), that's not great approach.

Adding these two configurations would be useful for tricky environment when, unfortunately, you can't manage your permissions yourself, or when multiple users try to access/edit these files.

Reply:
If they don't know how it works, they will touch and break all the thing because they will try to edit something to try to understand how it works.
This is a server configuration problem, if you need to have multiple users to play with your files, you are maybe on linux in development mode and you can configure your apache (**/etc/apache2/envars** for example) to use your own user/group instead **www-data**.

More, all FTP tools allow you to configure permissions and if you are not using an FTP, this is because you know how to configure a server, so, you should be able to set permissions and the new code will not override it.

We must not add system configuration, even wordpress has a good article about that: https://codex.wordpress.org/Changing_File_Permissions

Reply:
Yeas I know you can change the group from apache. That's the kind of usecase I was thinking about. But in that case to allow write permissions between a group, should we change the default permission to 775 and 664?

Reply:
If you have the same group yes, but the file will be created with the running apache user. And it becomes a complicated configuration which is not a good practice (not at all).

Reply:
ok then, let's keep it simple for now